src/MainMenu.cpp

```diff
@@ -113,3 +117,172 @@ ZunResult MainMenu::LoadTitleAnm(MainMenu *menu)
     return ZUN_SUCCESS;
 }
 #pragma optimize("", on)
+
+#pragma optimize("s", on)
+#pragma var_order(i, drawVm)
+ZunResult MainMenu::DrawStartMenu(void)
+{
+    int i;
+    i = MoveCursor(this, 8);
+    if ((this->cursor == 1) && !g_GameManager.hasReachedMaxClears(0, 0) && !g_GameManager.hasReachedMaxClears(0, 1) &&
+        !g_GameManager.hasReachedMaxClears(1, 0) && !g_GameManager.hasReachedMaxClears(1, 1))
+    {
+        this->cursor += i;
+    }
+    AnmVm *drawVm = this->vm;
+    for (i = 0; i < 8; i++, drawVm++ /* zun why */)
+    {
+        DrawMenuItem(drawVm, i, this->cursor, 0xffff0000, 0x80300000, 122);
+    }
+    if (this->stateTimer >= 0x14)
+    {
+        if (WAS_PRESSED(0x1001))
+        {
+            switch (this->cursor)
+            {
+            case 0:
+                for (i = 0; i < 122; i++)
+                {
+                    this->vm[i].pendingInterrupt = 4;
+                }
+                this->gameState = STATE_DIFFICULTY_LOAD;
+                g_GameManager.unk_1823 = 0;
+                if (EXTRA <= g_GameManager.difficulty)
+                {
+                    g_GameManager.difficulty = NORMAL;
+                }
+                if (EXTRA <= g_Supervisor.cfg.defaultDifficulty)
+                {
+                    g_Supervisor.cfg.defaultDifficulty = NORMAL;
+                }
+                this->stateTimer = 0;
+                this->unk_81fc = 0x40000000;
+                this->maybeMenuTextColor = 0xff000000;
+                this->unk_820c = 0;
+                this->isActive = 60;
```

Comment:
Hmm, this name is a bit unfortunate. I usually expect a variable named `isX` to be a boolean, containing only 0/1. Do we know what this is used for, and could we find a better name for it? If not, that's fine, we can always delegate this to the future.

Reply:
That's the funny thing, even thought it's not a bool, it's only ever set to either 60 or 0
the comment from the ghidra server (not by me) says

> Seems to only get set once the current menu is active (aka the function has been called), 0 when its not ready (loading, or on the title screen), and 60 when it is ready and active.

So like, the tldr is that it's not a bool de iure but it is de facto

Reply:
Alright, let's keep it that way then